python/hail/docs/hailpedia/matrixtable.rst

```diff
@@ -0,0 +1,404 @@
+
+Aggregation
+===========
+
+:class:`.MatrixTable` has three methods to compute aggregate statistics.
+
+- :meth:`.MatrixTable.aggregate_rows`
+- :meth:`.MatrixTable.aggregate_cols`
+- :meth:`.MatrixTable.aggregate_entries`
+
+These methods take an aggregated expression and evaluate it, returning
+a Python value.
+
+An example of querying entries is to compute the global mean of field `GQ`:
+
+    >>> mt.aggregate_entries(hl.agg.mean(mt.GQ))
+    67.73196915777027
+
+It is possible to compute multiple values simultaneously (and encouraged,
+because grouping two computations together will run twice as fast!) by
+creating a tuple or struct:
+
+    >>> mt.aggregate_entries((agg.stats(mt.DP), agg.stats(mt.GQ)))
+    (Struct(mean=41.83915800445897, stdev=41.93057654787303, min=0.0, max=450.0, n=34537, sum=1444998.9999999995),
+    Struct(mean=67.73196915777027, stdev=29.80840934057741, min=0.0, max=99.0, n=33720, sum=2283922.0000000135))
+
+See the :ref:`sec-aggregators` page for the complete list of aggregator
+functions.
+
+Group-By
+========
+
+Matrix tables can be aggregated along the row or column axis to produce a new
+matrix table.
+
+- :meth:`.MatrixTable.group_rows_by`
+- :meth:`.MatrixTable.group_cols_by`
+
+First let's add a random phenotype as a new column field `case_status` and then
+compute statistics about the entry field `GQ` for each grouping of `case_status`.
+
+    >>> mt_ann = mt.annotate_cols(case_status = hl.cond(hl.rand_bool(0.5),
+    ...                                                 "CASE",
+    ...                                                 "CONTROL"))
+
+Next we group the columns by `case_status` and aggregate:
+
+    >>> mt_grouped = (mt_ann.group_cols_by(mt_ann.case_status)
+    ...                 .aggregate(gq_stats = agg.stats(mt_ann.GQ)))
+
+    >>> print(mt_grouped.entry.dtype.pretty())
+    struct {
+        gq_stats: struct {
+            mean: float64,
+            stdev: float64,
+            min: float64,
+            max: float64,
+            n: int64,
+            sum: float64
+        }
+    }
+
+    >>> print(mt_grouped.col.dtype)
+    struct{status: str}
+
+Joins
+=====
+
+Joins on two-dimensional data are significantly more complicated than joins
+in one dimension, and Hail does not yet support the full range of
+joins on both dimensions of a matrix table.
+
+:class:`.MatrixTable` has methods for concatenating rows or columns:
+
+- :meth:`.MatrixTable.union_cols`
+- :meth:`.MatrixTable.union_rows`
+
+:meth:`.MatrixTable.union_cols` joins matrix tables together by performing an
+inner join on rows while concatenating columns together (similar to `paste` in
+Unix). Likewise, :meth:`.MatrixTable.union_rows` performs an inner join on
+columns while concatenating rows together (similar to `cat` in Unix).
+
+In addition, Hail provides support for joining data from multiple sources together
+if the keys of each source are compatible (same order and type, but the names do
```

Comment:
"order" is very unclear here, the fact that a "key" is a composition of several fields is not really clear from any of the prior verbiage. Hmm. It's hard to describe this. maybe: "compatible (meaning that, viewing the key as an ordered tuple of fields, those fields must have the same type, but need not have the same field names)"